doc/classes/NavigationPathQueryParameters2D.xml

```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="NavigationPathQueryParameters2D" inherits="RefCounted" version="4.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+	<brief_description>
+		Parameters to be sent to a 2D navigation path query.
+	</brief_description>
+	<description>
+		This class contains the start and target position and other parameters to be used with [method NavigationServer2D.query_path].
+	</description>
+	<tutorials>
+	</tutorials>
+	<members>
+		<member name="calc_path_length" type="bool" setter="set_calc_path_length" getter="get_calc_path_length" default="false">
+			If [code]true[/code], the length of the path will be calculated as the sum of the vector lengths between the path points.
+		</member>
+		<member name="map" type="RID" setter="set_map" getter="get_map">
+			The navigation [code]map[/code] [RID] used in the path query.
+		</member>
+		<member name="navigation_layers" type="int" setter="set_navigation_layers" getter="get_navigation_layers" default="1">
+			The navigation layers the query will use (as a bitmask).
+		</member>
+		<member name="optimize" type="bool" setter="set_optimize" getter="get_optimize" default="true">
+			If [code]true[/code], the path receives post-processing with a funnel algorithm along the polygon corridor create by the pathfinding to smooth the path. If [code]false[/code] all path positions will be placed at the center of each passed navigation mesh polygon edge.
+		</member>
+		<member name="pathfinding_algorithm" type="int" setter="set_pathfinding_algorithm" getter="get_pathfinding_algorithm" enum="NavigationPathQueryParameters2D.PathfindingAlgorithm" default="0">
+			The pathfinding algorithm used in the path query.
+		</member>
+		<member name="start_position" type="Vector2" setter="set_start_position" getter="get_start_position" default="Vector2(0, 0)">
+			The pathfinding start position in global coordinates.
+		</member>
+		<member name="strip_margin" type="float" setter="set_strip_margin" getter="get_strip_margin" default="0.001">
+			Margin value for normalized vector dot product to decide if intermediate path positions form enough of a straight line to be stripped. Used if [member strip_path] is [code]true[/code].
+		</member>
```

Comment:
This naming is not great for a dot product. But I suspect the name "margin" could be kept, but it should actually be a distance in world space for deviation from the path.

Although you may want to also have a param for tolerance of equal points so if this is not hard coded, these two names should make sense in terms of each other.

Reply:
Naming is not great, I actuall don't know how to describe that better, and suggestion?

Reply:
Maybe try `path_angular_deviation`?

Reply:
The path simplifying was removed.